src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,46 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $daysGap;
+  width: ($cellWidth * 7) + ($calendarPadding * 2) + ($daysGap * 6);
+  padding: $calendarPadding;
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    box-sizing: border-box;
+    height: $cellHeight;
+    width: $cellWidth;
+    border: 1px solid #000;
+    background-color: #eee;
+    transition: transform 0.5s;
+
+    &:hover {
+      background-color: #ffbfcb;
+      cursor: pointer;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $day from 1 through 31 {
+    & :nth-child(#{$day})::before {
+      content: '#{$day}';
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+
+  $weekDays: (mon 0, tue 1, wed 2, thu 3, fri 4, sat 5, sun 6);
+
+  @each $weekDay, $index in $weekDays {
+    &--start-day-#{$weekDay} :first-child {
+      margin-left: $cellWidth * $index + $daysGap * $index;
+    }
+  }
```

Comment:
The calculation of the margin-left property is a bit complex. It would be better to calculate the width of a single cell with gaps and then multiply it by the index. This would make the code more readable and maintainable.